src/custom_instance_manager.ts

```diff
@@ -0,0 +1,132 @@
+    constructor(options: CustomInstanceManagerOptions) {
+        this.isDryRun = options.isDryRun;
+
+        this.launchInstances = this.launchInstances.bind(this);
+    }
+
+    async launchInstances(
+        ctx: Context,
+        group: InstanceGroup,
+        groupCurrentCount: number,
+        quantity: number,
+    ): Promise<Array<string | boolean>> {
+        ctx.logger.info(`[custom] Launching a batch of ${quantity} instances in group ${group.name}`);
+
+        const indexes: Array<number> = [];
+        for (let i = 0; i < quantity; i++) {
+            indexes.push(i);
+        }
+
+        const result = await Promise.all(
+            indexes.map(async (index) => {
+                return this.launchInstance(ctx, index, group);
+            }),
+        );
+        ctx.logger.info(`Finished launching all the instances in group ${group.name}`);
+
+        return result;
+    }
+
+    async launchInstance(ctx: Context, index: number, group: InstanceGroup): Promise<string | boolean> {
+        const groupName = group.name;
+        const groupInstanceConfigurationId = group.instanceConfigurationId;
+
+        const displayName = groupName + '-' + makeRandomString(5);
+
+        ctx.logger.info(`[custom] Launching instance number ${index + 1} in group ${groupName} with properties`, {
+            groupName,
+            displayName,
+            groupInstanceConfigurationId,
+        });
+
+        if (this.isDryRun) {
+            ctx.logger.info(`[custom] Dry run enabled, skipping the instance number ${index + 1} launch`);
+            return true;
+        }
+        try {
+            const launchResponse = await this.execLaunch({
+                displayName,
+                groupName,
+                region: group.region,
+                type: group.type,
+            });
+            ctx.logger.info(
+                `[custom] Got launch response for instance number ${
+                    index + 1
+                } in group ${groupName}: ${launchResponse}`,
+            );
+
+            return launchResponse;
+        } catch (err) {
+            ctx.logger.error(
+                `[custom] Failed launching instance number ${index + 1} in group ${groupName} with err ${err}`,
+                { err },
+            );
+            return false;
+        }
+    }
+
+    async getInstances(): Promise<CustomCloudInstance[]> {
+        // TODO we might need another script to fetch the current instance status
+        return Object.values(this.instances);
+    }
+
+    execLaunch({
+        displayName,
+        groupName,
+        region,
+        type,
+    }: {
+        displayName: string;
+        groupName: string;
+        region: string;
+        type: string;
+    }): Promise<string> {
+        return new Promise(function (resolve, reject) {
+            exec(
+                `./scripts/custom-launch.sh --type ${type} --name ${displayName} --groupName ${groupName} --region ${region} `,
```

Comment:
I think the location of the script should be configurable, and have as a default the ./scripts/custom-launch.sh value